index.html

```diff
@@ -204,55 +204,1951 @@ <h3>
+          key, or `false` if the item creates variable data fields in reports
+          where each data field contains a value.
+        </dd>
+        <dt>
+          <dfn>isBufferedBytes</dfn> member
+        </dt>
+        <dd>
+          `true` if the item emits a fixed-sized stream of bytes, or `false` if
+          the item is a numeric quantity.
+        </dd>
+        <dt>
+          <dfn>isConstant</dfn> member
+        </dt>
+        <dd>
+          `true` if the item is a static read-only field and cannot be
+          modified, or `false` if the item defines report fields that contain
+          modifiable device data.
+        </dd>
+        <dt>
+          <dfn>isLinear</dfn> member
+        </dt>
+        <dd>
+          `true` if the item represents a linear relationship between what is
+          measured and what is reported, or `false` if the data has been
+          processed in some way.
+        </dd>
+        <dt>
+          <dfn>isRange</dfn> member
+        </dt>
+        <dd>
+          `true` if the item assigns usages from a [=HID Usage=] range defined
+          by {{HIDReportItem/usageMinimum}} and {{HIDReportItem/usageMaximum}},
+          or `false` if the item has a sequence of [=HID Usage=] values in
+          {{HIDReportItem/usages}}.
+        </dd>
+        <dt>
+          <dfn>isVolatile</dfn> member
+        </dt>
+        <dd>
+          <p>
+            `true` if the item value can change without host interaction, or
+            `false` if the item value should only be changed by the host.
+          </p>
+          <p>
+            Only used for Feature and Output items.
+          </p>
+        </dd>
+        <dt>
+          <dfn>hasNull</dfn> member
+        </dt>
+        <dd>
+          `true` if the item has a null state in which it is not sending
+          meaningful data, or `false` if it does not have a null state. When in
+          a null state, the item will report a value outside of the specified
+          {{HIDReportItem/logicalMinimum}} and
+          {{HIDReportItem/logicalMaximum}}.
+        </dd>
+        <dt>
+          <dfn>hasPreferredState</dfn> member
+        </dt>
+        <dd>
+          `true` if the item has a preferred state to which it will return when
+          the user is not physically interacting with the control, or `false`
+          if the item does not have a preferred state.
+        </dd>
+        <dt>
+          <dfn>wrap</dfn> member
+        </dt>
+        <dd>
+          `true` if the item value rolls over when reaching either the extreme
+          high or low value, or `false` if the item value does not roll over.
+        </dd>
+        <dt>
+          <dfn>usages</dfn> member
+        </dt>
+        <dd>
+          <p>
+            If {{HIDReportItem/isRange}} is `true` or this item has no
+            associated [=HID Usage=] values, then {{HIDReportItem/usages}} MUST
+            be an empty sequence.
```

Comment:
Since this is a dictionary and `isRange` is optional this could also be undefined.

Reply:
It looks like the algorithm to initialize {{HIDReportItem}} would leave `isRange` undefined if the item does not have a usage range, I've fixed it to set `isRange` to `false` in that case so now it should always be defined.

Web IDL recommends not using required for types that are only used as return values, so I've left it as optional.